### PR TITLE
Hooking up with React Hooks

### PR DESCRIPTION
**Pull request by JoaoTMDias**
Hooking up with React Hooks
=================================================

-   Speakers : João Dias and Mickael Costa
-   Available : first day and second day
-   Length : 90 minutes
-   Language : English

Description
-----------

In the last months, React Hooks changed how we write and think about reusable and straightforward code. This training workshop will cover the fundamentals of React Hooks, how to convert your codebase and also how to take advantage and create your own custom ones. All of this in simple, straightforward and understandable examples.

Speaker Bio
-----------

João is a Frontend Engineer at Feedzai with over 4 years of developer experience and over 9 as an interface designer. Mostly, he just loves to design and build stuff for the Web. His designer/developer cross-breed skills are his main strength and help him to bring the best of both worlds to every project he gets involved into.

Mickael is Senior Frontend Engineer at Feedzai. He had the chance to work and experiment with a variety of languages and frameworks namely React. He was a teacher so he enjoys teaching and learning with a good conversation about technology. He has other tastes like sports (futebol and Krav Maga) and travel.

Links
-----

João:

-   Company: https://feedzai.com
-   GitHub: https://github.com/joaotmdias
-   Personal: https://joaodias.me
-   LinkedIn: https://www.linkedin.com/in/joaotmdias/

Mickael:

-   Company: https://feedzai.com
-   GitHub: https://github.com/Mickael24
-   LinkedIn: https://www.linkedin.com/in/dinomickael/

Extra Information
-----------------

João

-   [Speaker at ENEMM 2019 with the keynote "State of the Web in 2019"](https://www.enemm.pt/speakers/joao-dias)
-   _Workshop: The User's Perception of Speed_ (2016)
-   _Keynote: State of Javascript 2018: Angular or Vue or React?_ (2018)
-   _Keynote: Let's talk about Accessibility_ (2019)

Mickael

-   ["First React Application" workshop teacher at FEUP Code Week 2019"](https://www.linkedin.com/posts/dinomickael_feup-codeweek-feedzai-activity-6592090630468448256-k9Jk/)
-   Assistant Professor at the Polytechnic Institute of Cávado and Ave

By the end of this workshop attendees should be able to do the following:

-   Use built-in React Hooks to create more reusable and straightforward code
-   Re-factor Class components into functional components using hooks
-   Create custom hooks and reuse them throughout a project
-   Know where React Hooks fall short and we should choose class components instead.

### CODE REVIEW

**Comment by carlosefr** [comment]
Hi, can you please fix the markup on your `.md` file to match the one used on the template?

**Comment by JoaoTMDias** [comment]
Hi Carlos! Yes, I'll edit this and re-submit again. Thanks!

**Comment by JoaoTMDias** [comment]
> Hi, can you please fix the markup on your `.md` file to match the one used on the template?

Ok, I think now it's ok for review.
If you find anything else that needs change, please say so 🙂

**Comment by carlosefr** [comment]
Thanks.

On another note: From the description, this looks like a talk proposal. What makes this proposal a workshop, besides being longer than one hour?

**Comment by JoaoTMDias** [comment]
> Thanks.
> 
> On another note: From the description, this looks like a talk proposal. What makes this proposal a workshop, besides being longer than one hour?

Well, that is a good question! 🙂

On one hand, we feel that a talk has a very short amount of time just for us to go through (even simple) examples for baked-in hooks like useState, useEffect, useReducer, etc.

On the other, we need more time to go through those first and then explain how to convert a class-based project to hooks and take advantage also of custom hooks to create cleaner code.

So the idea of this as a workshop is to go through separate sections (ex: what they are, how to "think in hooks", what are the basics, how to create custom hooks, etc ), have exercises at the end of each section and have people asking questions. People will probably ask questions about testing hooks, others about using hooks with typescript, so we would also need time to address those, if necessary.

Basically, there are tons of articles about React hooks out there, and in our view, the workshop is a better approach for people to sit, listen, but also write down and practice. 🙂

**Comment by carlosefr** [comment]
Ok, I think I understand. :)

**Comment by carlosefr** [comment]
Congratulations @JoaoTMDias, your talk proposal has been pre-accepted.

This means it's actually approved for Pixels Camp just not yet in our scheduling workflow (that will happen at a later date, and you should get an email about it when it happens).

In the meantime, if you have any questions, you can contact us at org@pixels.camp.

**Please note:** I can't find @Mickael24 in our attendee database. Both speakers must apply and be accepted as attendees to enter the venue.

**Comment by Mickael24** [comment]
@carlosefr 
I had already submitted it, but for some reason, it was not correct. Can you check now?

Thanks for the opportunity :)

**Comment by carlosefr** [comment]
All OK now. :)

**Comment by JoaoTMDias** [comment]
Sorry, @carlosefr can we switch the availability only for the first and second days?

**Comment by carlosefr** [comment]
We're sad to inform you that Pixels Camp v4.0 has been canceled due to the still ongoing pandemic. Please refer to this blog post for more details:

https://blog.pixels.camp/about-pixels-camp-v4-0-6fd8b91cc8f2

On behalf of the organization, thank you for having proposed a talk. We would have been delighted to have had you with us in Lisbon in November to deliver it and hope you'll consider submitting again for a future Pixels Camp edition.
